src/pelicanfs/token_content_iterator.py

```diff
@@ -82,21 +96,203 @@ class TokenContentIterator:
         4. Default token file via default_bearer_token_file()
         5. Environment variable TOKEN (interpreted as file path)
         6. HTCondor discovery via _CONDOR_CREDS or .condor_creds directory
+        7. OIDC device flow via pelican binary (final fallback)
 
     Attributes:
         location (str): Specific token file path (optional).
         name (str): Logical name of the token (used by HTCondor discovery).
+        operation: Token operation type (read/write).
+        destination_url (str): Destination URL for the token request.
+        pelican_url (str): Pelican protocol URL (pelican://<federation-url>/<path>) for OIDC device flow.
         method_index (int): Internal index of the current discovery method.
         cred_locations (List[str]): Token file paths discovered via HTCondor fallback.
         index (int): Internal index of the current fallback cred_location
     """
 
     location: str
     name: str
+    operation: Optional[object] = None
+    destination_url: Optional[str] = None
+    pelican_url: Optional[str] = None
     method_index: int = 0
     cred_locations: List[str] = field(default_factory=list)
     fallback_index: int = 0
 
+    def _pelican_binary_exists(self) -> bool:
+        """Check if pelican binary exists in PATH"""
+        return shutil.which("pelican") is not None
```

Comment:
It could be helpful to add a debug statement that displays the contents of PATH this invocation is considering.